src/algoliasearch.js

```diff
@@ -120,6 +120,8 @@ var AlgoliaSearch = function(applicationID, apiKey, methodOrOptions, resolveDNS,
     }
 };
 
+AlgoliaSearch.JSONPCounter = 0;
```

Comment:
This will be used as part the JSONP id, so I wonder if this would be better to name it JSONP_ID?

Will add a comment.